misk-core/src/main/kotlin/misk/compress/DeflateCompressionCodec.kt

```diff
@@ -0,0 +1,52 @@
+package com.squareup.misk.compress
+
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.util.zip.Deflater
+import java.util.zip.Deflater.DEFAULT_COMPRESSION
+import java.util.zip.DeflaterOutputStream
+import java.util.zip.InflaterOutputStream
+
+/**
+ * A {@link CompressionCodec} implementation for the deflate compression format. It uses the
+ * implementation provided in the standard {@code java.util.zip} package.
+ *
+ * Cribbed from SQ/repos/java/browse/compression.
+ */
+class DeflateCompressionCodec(private val compressionLevel: Int = DEFAULT_COMPRESSION) :
+    CompressionCodec {
+  init {
+    check(-1 <= compressionLevel && compressionLevel <= 9) {
```

Comment:
require –> you passed a bad argument
check –> you can’t call this now

Reply:
> I think I’d prefer `@SnappyCompressed` or `@DeflateCompressed` rather than fully pluggable compression codecs. The world market for compression algorithms is near to 1. We probably want Deflate and perhaps Brotli.

That seems reasonable – note that I've not really locked in the annotations yet